claims_db/phclaims/stage/tables/load_stage.mcare_mbsf.r

```diff
@@ -0,0 +1,497 @@
+## Header ####
+    # Author: Danny Colombara
+    # 
+    # R version: 3.5.3
+    #
+    # Purpose: Simple processing of load_raw MBSF to staged MBSF
+    # 
+    # Notes: Medicare ids are case sensitive. Will have to deduplicate randomly (without regard to case sensitivity) becuase there
+    #        is no way to know which one we should keep for linkages with other datasets. Checked on 8/15/2019 and found that there 
+    #        were 378 IDs that would be dropped through this process
+    #
+    #        This code takes approximately 32 minutes to run under normal SQL server traffic conditions
+
+## Set up environment ----
+    rm(list=ls())
+    pacman::p_load(data.table, dplyr, odbc, lubridate, glue, httr)
+    start.time <- Sys.time()
+    options("scipen"=999) # turn off scientific notation  
+    options(warning.length = 8170) # get lengthy warnings, needed for SQL
+    setwd("C:/temp/")
+    yaml.ab.url <- "https://raw.githubusercontent.com/PHSKC-APDE/claims_data/master/claims_db/phclaims/stage/tables/create_stage_mcare_mbsf_ab.yaml"
+    yaml.abcd.url <- "https://raw.githubusercontent.com/PHSKC-APDE/claims_data/master/claims_db/phclaims/stage/tables/create_stage_mcare_mbsf_abcd.yaml"
+    
+## (1) Connect to SQL Server ----    
+    db_claims <- dbConnect(odbc(), "PHClaims51") 
+
+## (2) PULL & PREP MBSF AB ----
+    # count rows in load_raw 
+    ab.raw.count <- as.numeric(odbc::dbGetQuery(db_claims, 
+                                             "SELECT COUNT (*) FROM load_raw.mcare_mbsf_ab"))
+    # pull load_raw into R
+    ab <- (
+      odbc::dbGetQuery(db_claims, 
+                       "SELECT DISTINCT * FROM load_raw.mcare_mbsf_ab"))
+    setDT(ab)
+    
+    # ensure there are no errors when pulling data from SQL
+    if(ab.raw.count != nrow(ab))
+      stop("Mismatching row count, error reading in data")
+    
+    # Add date stamp to data
+    ab[, last_run := Sys.time()]
+    
+    # read in YAML data for new stage file from GitHub
+    x <- GET(yaml.ab.url, authenticate(Sys.getenv("GITHUB_TOKEN"), ""))
+    stop_for_status(x)
+    x <- content(x, type="text", encoding = "ISO-8859-1")
+    writeLines(x, con="temp.yaml")
+    table_config <- yaml::read_yaml("temp.yaml")
+    file.remove("temp.yaml")
+    
+    # identify the column types needed for stage file in SQL
+    ab.stage.cols <- data.table(col.name = names(table_config$vars), col.type = table_config$vars)
+    ab.characters <- ab.stage.cols[tolower(col.type) %like% "varchar", ]$col.name
```

Comment:
Suggest checking for char rather than varchar in case you have some fixed-wdith columns

Reply:
agreed.  just made the change in the code.